### ISSUE TITLE
Karma v6 breaking changes

### ISSUE THREAD

**Issue by devoto13** [karma-runner/karma#3503]
This issue is to track breaking changes for Karma v6 release. It will be updated with more points as I discover them.

- [ ] Enable [`strict` mode](https://yargs.js.org/docs/#api-strictenabledtrue) for `yargs` to error out when unknown command line options are provided

**Comment by devoto13** [comment]
@johnjbarton Since we merged #3586 with the breaking change, next release will be 6.0.0. What do you think about the points described in this issue? I'm thinking to start opening PRs for them, ending with "Drop support for legacy browsers", which should fix CI on master (currently fails in Safari 9) and publish 6.0.0 containing all the breaking changes in one release.

**Comment by johnjbarton** [comment]
Some of these would be great. The `ws` one seems like it could take a bit to get right.  Let me check on IE<11.

At present it seems like our semantic release is broken. So to prevent a release I need to stop trying to get it work or otherwise disable it.  I'll look in to the latter.  

I don't want to wait too long however.

**Comment by devoto13** [comment]
No problems. I'll try to get PRs up in the coming days.

**Comment by devoto13** [comment]
@johnjbarton Created a PR for deprecated API removal (actually there were not so much to remove 😄). 

Everything else described in this issue depends on the dropping of legacy browsers support, so I'll wait for the confirmation that we can actually drop this support before I start implementing the rest.

**Comment by johnjbarton** [comment]
I think we should stick with the shorter list for v6, including the reconnect changes I am working on. We can then start v7 immediately focusing on the IE<11 code. Any v6 issues will be cleared up then.

**Comment by johnjbarton** [comment]
Ok I think that when I land #3598 the releases will start. If not then I will focus on why not. 
Plan to land #3601 and #3585 first, after reviews.

**Comment by jehon** [comment]
If you land https://github.com/karma-runner/karma/pull/3605 too, you can say that v6 will have full module support... :-)

**Comment by johnjbarton** [comment]
I believe that #3609 will cause the release to happen.  But then again I said that before....

**Comment by XhmikosR** [comment]
I was looking at the dependencies, and maybe `http2` is redundant too?

**Comment by devoto13** [comment]
@XhmikosR Yes, I agree. See "Remove support for http2 module."` checkbox in the issue.
